unified_planning/io/pddl_reader.py

```diff
@@ -357,22 +369,66 @@ def parse_problem(self, domain_filename: str,
                             raise SyntaxError(f'Not able to handle duration constraint of action {n}')
                     if lower is None or upper is None:
                         raise SyntaxError(f'Not able to handle duration constraint of action {n}')
-                    d = unified_planning.model.ClosedIntervalDuration(self._em.Real(lower),
-                                                         self._em.Real(upper))
+                    d = up.model.ClosedIntervalDuration(self._em.Real(lower),
+                                                        self._em.Real(upper))
                     dur_act.set_duration_constraint(d)
                 else:
                     raise SyntaxError(f'Not able to handle duration constraint of action {n}')
                 cond = a['cond'][0]
                 self._add_condition(problem, dur_act, cond)
                 eff = a['eff'][0]
                 self._add_timed_effects(problem, dur_act, eff)
+
+                if has_actions_cost:
+                    if totalcost in fve.get(dur_act.duration().lower()) or \
+                       totalcost in fve.get(dur_act.duration().upper()):
+                        has_actions_cost = False
+                if has_actions_cost:
+                    for _, cl in dur_act.conditions().items():
+                        for c in cl:
+                            if totalcost in fve.get(c):
+                                has_actions_cost = False
+                                break
+                if has_actions_cost:
+                    for _, cl in dur_act.durative_conditions().items():
+                        for c in cl:
+                            if totalcost in fve.get(c):
+                                has_actions_cost = False
+                                break
+                if has_actions_cost:
+                    for _, el in dur_act.effects().items():
+                        for e in el:
+                            if totalcost in fve.get(e.fluent()) or totalcost in fve.get(e.value()) \
+                               or totalcost in fve.get(e.condition()):
+                                has_actions_cost = False
+                                break
```

Comment:
If you wrap this code in a function, you can use `return` to avoid the repeated `if has_actions_cost:`